.buildkite/scripts/lib/manifest.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Results in a single sha256sum of all files in directory
+sha256_of_dir() {
+    local -r dir_path="${1}"
+    find "${dir_path}" -type f -exec sha256sum {} \; | sha256sum | awk '{print $1;}'
+}
+
+manifest_contents() {
+    local -r version="${1}"
+    jq --null-input \
+        --arg version "${version}" \
+        '{"version": $version}'
```

Comment:
I trust you are anticipating additional content for this manifest in the near future? Otherwise, this `jq` invocation is _way_ overkill.

Reply:
Also, calling this a "manifest" is a bit misleading, since that generally implies some kind of enumeration of contents.

The manifest could end up being something like a structured version of `find "${dir}" -type f | sort | xargs sha256sum`, and then the final checksum would just be a checksum of _that file_.

(But even then, that's just a listing of the _source_ files, and not a listing of the files in the final generated artifact (along with their individual checksums)... so it's more of an "input manifest".)

Reply:
Changed it to 'artifact-metadata.' And yes, I correctly anticipated additional content in this manifest!

Reply:
![](https://media.giphy.com/media/W3a0zO282fuBpsqqyD/giphy.gif)